pallets/auctions/src/lib.rs

```diff
@@ -363,8 +366,13 @@ pub mod pallet {
 		/// Creates a new auction for a given Auction type
 		///
 		/// - calls the create() implementation on the given Auction type
-		/// 
-		#[pallet::weight(<T as Config>::WeightInfo::create_auction())]
+		///
+		#[pallet::weight(
+			<T as Config>::WeightInfo::create_english()
+				.max(<T as Config>::WeightInfo::create_topup())
+				.max(<T as Config>::WeightInfo::create_candle())
+		)]
```

Comment:
i would probably go with parametrize create_auction weight function where the create_auction would take a type of auction. 

In that case, you could actually return correct weight and not the max weight .

same could be done for all other extrinsics?